client.go

```diff
@@ -178,25 +180,35 @@ func (c Client) findIDInConfigurationsPayload(body io.ReadCloser, name string) (
 	return "", err
 }
 
+// ListNodes returns a list of dataset agent nodes from Flocker Control Service
+func (c *Client) ListNodes() (nodes []NodeState, err error) {
+	resp, err := c.get(c.getURL("state/nodes"))
+	if err != nil {
+		return []NodeState{}, err
+	}
```

Comment:
We should return an error for HTTP response codes >=300 for consistency with the other methods and so that the JSON decoding doesn't blow up with the non-json response body that you get from 500 errors.